quipucords/scanner/openshift/entities.py

```diff
@@ -155,29 +158,70 @@ def __raise__(self):
         return (self.message,)
 
 
-# update nested model references - this should always be the last thing to run
-_update_model_refs()
-
-
-@dataclass
-class Cpu:
-    """Class representing CPU's attributes."""
+class NodeResources(OCPBaseEntity):
+    """Class representing node's resources."""
+
+    cpu: float = None
+    memory_in_bytes: int = None
+    pods: int = None
+    _kind = "node-resources"
+
+    # https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu
+    @validator("cpu", pre=True)
+    def _convert_cpu_value(cls, value):
+        millicore_suffix = re.compile("^\d+m$")
+        if isinstance(value, str) and millicore_suffix.match(value):
+            value = float(value.replace("m", "")) / 1000
+        else:
+            raise ValueError(CPU_VALUE_ERROR)
+        return value
+
+    # https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
+    @validator("memory_in_bytes", pre=True)
+    def _convert_memory_bytes(cls, value):
+        if isinstance(value, str):
+            binary_unit_suffix = re.compile("^\d+([KMGTPE]i)$")
+            group_match_1 = binary_unit_suffix.match(value)
+
+            fixed_point_suffix = re.compile("^\d+([KMGTPE])$")
+            group_match_2 = fixed_point_suffix.match(value)
+            if group_match_1:
+                value = convert_memory_binary_unit(value, group_match_1)
+            elif group_match_2:
+                value = convert_memory_fixed_point(value, group_match_2)
+            else:
+                raise ValueError(MEMORY_CONVERSION_ERROR)
+        else:
+            raise ValueError(MEMORY_VALUE_ERROR)
```

Comment:
I don't think we need to raise an error in this case. Consider we are deserializing an already parsed result with this entity.

Reply:
Your jugment is correct in both cases. I will modify.